src/server/memory/usage.h

```diff
@@ -320,9 +444,56 @@ class ColdObjectTracker
 
   Status OnDelete(ID const& id) { return Self().OnDelete(id); }
 
+ protected:
+  Status SpillPayload_(std::shared_ptr<P>& payload) {
+    assert(payload->is_sealed);
+    util::SpillWriteFile write_file(spill_path);
+    RETURN_ON_ERROR(write_file.Write(payload));
+    RETURN_ON_ERROR(write_file.Sync());
+    BulkAllocator::Free(payload->pointer, payload->data_size);
+    payload->store_fd = -1;
+    payload->pointer = nullptr;
+    payload->is_spilled = true;
+    return Status::OK();
+  }
+
+  Status ReloadPayload_(const ID& id, std::shared_ptr<P>& payload) {
+    assert(payload->is_spilled == true);
+    util::SpillReadFile read_file(spill_path);
+    RETURN_ON_ERROR(read_file.Read(payload, Self().shared_from_this()));
+    return Status::OK();
+  }
+
+  Status DeletePayloadFile_(const ID& id) {
+    util::FileIOAdaptor io_adaptor(spill_path);
+    RETURN_ON_ERROR(io_adaptor.RemoveFile(spill_path + std::to_string(id)));
+    return Status::OK();
+  }
+
+  void SetSpillPath(const std::string& spill_path) {
+    this->spill_path = spill_path;
+    if (spill_path.empty()) {
+      LOG(ERROR) << "No spill path set, disable spill...";
+      return;
+    }
+    if (spill_path.back() != '/') {
+      this->spill_path.push_back('/');
+    }
+    util::FileIOAdaptor io_adaptor(this->spill_path + "test");
+    if (io_adaptor.Open("w").ok()) {
+      io_adaptor.RemoveFile(this->spill_path + "test");
+    } else {
+      LOG(ERROR) << "No such directory " << this->spill_path;
+      LOG(ERROR) << "So disable spill...";
```

Comment:
Why this message is an "ERROR"?

Reply:
Cuz user enter a wrong filepath, they expect to enable spill but we couldn't do that.

Reply:
Or maybe i could change it to WARNING

Reply:
As you doesn't abort, a `WARNING` might be more accurate. I suggest

```
LOG(WARNING) << "Disabling spilling as the specified spill directory doesn't exist, or vineyardd doesn't have the permission to write it".
```